src/command.py

```diff
@@ -49,6 +49,43 @@ def __repr__(self):
                f"Gem message: {self.gem_message}"
 
 
+class GemsCounterFromMessage:
+    def __get_int_converted_gem_count(self, gem_count_string: str):
+        if gem_count_string.isdigit():
+            return int(gem_count_string)
+        try:
+            gem_count_float = float(gem_count_string)
+            return int(gem_count_float)
+        except ValueError:
+            return 0
+
+    def _get_gem_count_from_gem_string(self, message: str):
+        gem_message_string_anchor = '$gem-'
+        if message.count(gem_message_string_anchor) == 0:
+            return 0
+
+        gem_message_words = message.split()
```

Comment:
Could this not be done with regex?
We are iterating multiple times, discord does have a hard limit on the response time of its hooks. This Probably leads me to suggest if we should separate the gem option from the message option?

Reply:
that is an awesome idea :D implemented regex in commit [682d44d](https://github.com/CraftsmenLtd/DiscordGems/pull/30/commits/682d44df0c99efc675bebf8e3033d58f82061f4a) and it seems to have reduced the lines of code significantly.

i did consider regex at first, but was not aware that we can make a list of matching strings with regex. i was mistaken to think after matching with regex, i will have to use split and match string anyway to parse and get the number. of course that was a mistake, and learned something new :D

thank you mate for the great feedback :+1: